build-tools/cache_samples.sh

```diff
@@ -99,11 +99,27 @@ function cache_devfile() {
     fi
 }
 
-devfileEntriesFile=$1
-samplesDir=$2
+if [[ "$OSTYPE" == "darwin" ]]
+then
+  if [[ $1 != ./* ]] && [[ $1 != ../* ]] && [[ $1 != /* ]]
+  then
+    devfileEntriesFile=./$1
+  else
+    devfileEntriesFile=$1
+  fi
+  if [[ $2 != ./* ]] && [[ $2 != ../* ]] && [[ $2 != /* ]]
```

Comment:
```suggestion
  if [[ $2 != ./* ]] && [[ $2 != ../* ]]
```

If my `$samplesdir` was `/samples/.cache` for example it wasn't adding the `.` to make it `./samples/.cache` in this case

Reply:
> If my `$samplesdir` was `/samples/.cache` for example it wasn't adding the `.` to make it `./samples/.cache` in this case

@Jdubrick I think this should be the case though, if one explicitly specifies `/samples/.cache` it shouldn't assume you mean a relative path. Correct me if I'm misunderstanding.

Reply:
@michael-valdron my from understanding when I pass `/samples/.cache` as the argument from the working directory when the script tries to create the directory it gets blocked by the OS on Mac because it assumes you are creating the directory from root. It may just be a Mac limitation that it has to assume you mean relative to the working directory? wdyt?

Reply:
Actually, I guess it could just be passed as `./samples/.cache` in this case to combat that, I believe you're right

Reply:
> Actually, I guess it could just be passed as `./samples/.cache` in this case to combat that, I believe you're right

Can confirm this is the case, I think this is good in that case!